__tests__/app.test.js

```diff
@@ -97,7 +97,33 @@ describe('/api/articles', () => {
       });
   });
 });
-
+describe('/api/articles?topic', () => {
+  test('GET 200: responds only with articles of a given topic', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(12);
+        expect(body.articles[0].topic).toBe('mitch');
+      });
+  });
+  test('GET 400: sends an error when filtering by an invalid topic', () => {
+    return request(app)
+      .get('/api/articles?topic=not_mitch')
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('bad request');
+      });
+  });
+  test('GET 404: sends an error when topic is valid but empty', () => {
```

Comment:
The technology topic is not in the data - I'll jot some thoughts on your model in a moment. There is a topic of 'paper' which exists but has no associated articles. We'd expect our query to respond with an empty array and a 200 status.